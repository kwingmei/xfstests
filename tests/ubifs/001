#! /bin/bash
# FS QA Test 001
#
# This is a test for mkfs.ubifs, mkfs.ubifs accept a -r option
# to build a file system from a directory. This case is testing
# this option and check the result of it.
#
#-----------------------------------------------------------------------
# Copyright (c) 2015 Dongsheng Yang <yangds.fnst@cn.fujitsu.com>.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=`mktemp -d`
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
	cd /
	rm -rf $tmp
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter

# real QA test starts here
_supported_fs ubifs
_supported_os Linux
_require_test
_require_scratch
_require_fssum
_need_to_be_root

rm -f $seqres.full

rootdir=$tmp/root
mkdir $rootdir

for i in $(seq 1 10); do
	dd if=/dev/urandom of=$rootdir/$i bs=1M count=1 >> $seqres.full 2>&1 || _fail "dd failed"
done

run_check $FSSUM_PROG -A -f -w $tmp/fssum $rootdir

_scratch_mkfs "-r $rootdir" >/dev/null 2>&1
_scratch_mount

run_check $FSSUM_PROG -r $tmp/fssum $SCRATCH_MNT 2>> $seqres.full

# success, all done
status=0
exit
