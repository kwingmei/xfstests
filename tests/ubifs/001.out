QA output created by 001
